samples/sensor/bq34110/src/main.c

```diff
@@ -0,0 +1,160 @@
+		memset(&avg_power, 0, sizeof(avg_power));
+		memset(&int_temp, 0, sizeof(int_temp));
+		memset(&state_of_health, 0, sizeof(state_of_health));
+		memset(&time_to_full, 0, sizeof(time_to_full));
+		memset(&time_to_empty, 0, sizeof(time_to_empty));
+
+		printk("Observation: %d\n", observation);
+
+		ret = sensor_sample_fetch_chan(dev, SENSOR_CHAN_ALL);
+		if (ret < 0) {
+			printk("Unable to fetch the sensor data\n");
+		}
+
+		ret = sensor_channel_get(dev, SENSOR_CHAN_GAUGE_VOLTAGE,
+					 &voltage);
+		if (ret < 0) {
+			printk("Unable to get the Voltage value\n");
+		}
+
+		printk("Voltage: %d.%06dV\n", voltage.val1, voltage.val2);
+
+		ret = sensor_channel_get(dev, SENSOR_CHAN_GAUGE_AVG_CURRENT,
+					 &current);
+		if (ret < 0) {
+			printk("Unable to get the Current Value\n");
+		}
+
+		bq34110_show_values("Avg Current in Amps", current);
+
+		ret = sensor_channel_get(dev,
+					 SENSOR_CHAN_GAUGE_STATE_OF_CHARGE,
+					 &state_of_charge);
+		if (ret < 0) {
+			printk("Unable to get the State of Charge\n");
+		}
+
+		printk("State of Charge: %d%%\n", state_of_charge.val1);
+
+		ret = sensor_channel_get(dev,
+					 SENSOR_CHAN_GAUGE_STATE_OF_HEALTH,
+					 &state_of_health);
+		if (ret < 0) {
+			printk("Unable to get State of Charge\n");
+		}
+
+		ret = sensor_channel_get(dev, SENSOR_CHAN_GAUGE_AVG_POWER,
+					 &avg_power);
+		if (ret < 0) {
+			printk("Unable to get Avg Power\n");
+		}
+
+		bq34110_show_values("Avg Power in Watt", avg_power);
+
+		ret = sensor_channel_get(dev,
+			SENSOR_CHAN_GAUGE_FULL_CHARGE_CAPACITY,
+			&full_charge_capacity);
+		if (ret < 0) {
+			printk("Unable to get Full Charge Capacity\n");
+		}
+
+		printk("Full Charge Capacity: %d.%06dAh\n",
+		       full_charge_capacity.val1, full_charge_capacity.val2);
+
+		ret = sensor_channel_get(dev,
+				SENSOR_CHAN_GAUGE_REMAINING_CHARGE_CAPACITY,
+				&remaining_charge_capacity);
+		if (ret < 0) {
+			printk("Unable to get Remaining Charge Capacity\n");
+		}
+
+		printk("Remaining Charge Capacity: %d.%06dAh\n",
+		       remaining_charge_capacity.val1,
+		       remaining_charge_capacity.val2);
+
+		ret = sensor_channel_get(dev, SENSOR_CHAN_GAUGE_TEMP,
+					 &int_temp);
+		if (ret < 0) {
+			printk("Unable to read Internal Temperature\n");
+		}
+
+		printk("Gauge Temperature: %d.%06d C\n", int_temp.val1,
+		       int_temp.val2);
+
+		ret = sensor_channel_get(dev,
+					SENSOR_CHAN_GAUGE_TIME_TO_FULL,
+					&time_to_full);
+		if (ret < 0) {
+			printk("Unable to read Time to Full\n");
+		}
+
+		printk("Time to Full: %dminutes\n", time_to_full.val1);
+
+		ret = sensor_channel_get(dev,
+					SENSOR_CHAN_GAUGE_TIME_TO_EMPTY,
+					&time_to_empty);
+		if (ret < 0) {
+			printk("Unable to read Time to Empty\n");
+		}
+
+		printk("Time to empty: %dminutes\n", time_to_empty.val1);
+
+		k_sleep(K_MSEC(5000));
+
+		observation++;
+	}
+}
+
+void main(void)
+{
+	const struct device *dev;
+
+	dev = device_get_binding(DT_LABEL(DT_INST(0, ti_bq34110)));
```

Comment:
Please use `DEVICE_DT_GET_ANY(ti_bq34110)` to get the device, and check it with `device_is_ready`.